type.go

```diff
@@ -182,16 +194,31 @@ func parseRequestType(t reflect.Type) (*requestType, error) {
 			return nil, errgo.Mask(err)
 		}
 
-		if f.Anonymous {
-			if tag.source != sourceBody && tag.source != sourceNone {
-				return nil, errgo.New("httprequest tag not yet supported on anonymous fields")
-			}
+		if f.Anonymous && tag.source != sourceNone {
+			taggedFieldIndex = f.Index
 		}
 		pt.fields = append(pt.fields, field)
 	}
 	return &pt, nil
 }
 
+// withinIndex reports whether the field index i0 should be considered
+// to be within the field with index i1.
+func withinIndex(i0, i1 []int) bool {
+	// The index of a field within an anonymous field is formed by
```

Comment:
This is a public contract in reflect. Every field has an index within its struct. Fields in anonymous struct fields have multi-level indexes (one for the anonymous struct parent and one for the field itself).